sklearn/cluster/tests/test_birch.py

```diff
@@ -121,6 +122,35 @@ def test_partial_fit_second_call_error_checks():
     with pytest.raises(ValueError, match=msg):
         brc.partial_fit(X[:, [0]], y)
 
+def test_radius():
+    # Test that the radius is as defined, simple case
+    c = _CFSubcluster()
+    c.update(_CFSubcluster(linear_sum=np.array([0])))
+    c.update(_CFSubcluster(linear_sum=np.array([1])))
+    assert c.radius == 0.5, c.radius
```

Comment:
I think we should not expect exact equality between too floating point numbers in tests so instead the following would be more appropriate.

```suggestion
    assert c.radius == pytest.approx(0.5)
```

Also, I think the trailing `, c.radius` is not necessary because pytest will already report an informative error message in case of failure.

Reply:
Similar changes should be applied to all the other assert statements of this PR.

Reply:
There are no possible rounding differences here. Its 1 divided by 2.
0.5 is exact here, and should be the exact result independent of the architecture.
But feel free to change this to your liking if you have a fix for this bug.